src/main/java/org/takes/misc/Href.java

```diff
@@ -321,4 +292,53 @@ private static URI createURI(final String txt) {
         }
         return result;
     }
+
+    /**
+     * Convert the provided query into a Map.
+     * @param query The query to parse.
+     * @return A map containing all the query arguments and their values.
+     */
+    @SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
+    private static SortedMap<String, List<String>> asMap(final String query) {
+        final SortedMap<String, List<String>> params =
+            new TreeMap<String, List<String>>();
+        if (query != null) {
+            final String[] pairs = query.split("&");
```

Comment:
@essobedo I mean
the variable `pairs` used only once
so this code
```
final String[] pairs = query.split("&");
    for (final String pair : pairs) {
```
can be rewritten as
```
    for (final String pair : query.split("&")) {
```

also please check out  this article 
http://www.yegor256.com/2014/04/27/typical-mistakes-in-java-code.html#one-time-variables